src/api/server.rkt

```diff
@@ -252,26 +275,73 @@
                          *loose-plugins*)
    (parameterize ([current-error-port (open-output-nowhere)]) ; hide output
      (load-herbie-plugins))
+   (define worker-thread
+     (thread (λ ()
+               (let loop ([seed #f])
+                 (match (thread-receive)
+                   [`(init rand
+                           ,vec
+                           flags
+                           ,flag-table
+                           num-iters
+                           ,iterations
+                           points
+                           ,points
+                           timeout
+                           ,timeout
+                           output-dir
+                           ,output
+                           reeval
+                           ,reeval
+                           demo?
+                           ,demo?)
+                    (set! seed vec)
+                    (*flags* flag-table)
+                    (*num-iterations* iterations)
+                    (*num-points* points)
+                    (*timeout* timeout)
+                    (*demo-output* output)
+                    (*reeval-pts* reeval)
+                    (*demo?* demo?)]
+                   [job-info (run-job job-info)])
+                 (loop seed)))))
+   (define timeline (*timeline*))
```

Comment:
I don't think you need to read `(*timeline*)` here; setting it to `#f` might help catch bugs if you did the save wrong.